src/compiler/utilities.ts

```diff
@@ -1788,6 +1789,25 @@ namespace ts {
         return init && getExpandoInitializer(init, isPrototypeAccess(node.name));
     }
 
+    export function getDescriptorInitializerOfKind(props: NodeArray<ObjectLiteralElementLike>, kind: "value" | "get" | "set") {
+        const init = find(props, p => !!p.name && getTextOfPropertyName(p.name) === kind);
+        if (init) {
+            switch (init.kind) {
+                case SyntaxKind.PropertyAssignment:
+                    return init.initializer;
+                case SyntaxKind.ShorthandPropertyAssignment:
+                case SyntaxKind.MethodDeclaration:

```

Comment:
which ones are allowed to be methods? get? set? value?

Reply:
Get and set often are. Value can be, it'd be the same as a property with a function expression value, though.

Reply:
I think this function is unused in my current implementation, though.